lake/commit/snapshot.go

```diff
@@ -19,6 +18,7 @@ var (
 type View interface {
 	Lookup(ksuid.KSUID) (*segment.Reference, error)
 	Select(extent.Span, order.Which) Segments
+	SelectAll() Segments
```

Comment:
Doesn't have to happen now but `SelectAll` should go away since it's equivalent to `Select` with a nil `extent.Span`.